utils/funcexp/func_math.cpp

```diff
@@ -491,6 +547,21 @@ double Func_cos::getDoubleVal(Row& row,
         }
         break;
 
+        case execplan::CalpontSystemCatalog::LONGDOUBLE:
+        {
+            // null value is indicated by isNull
+            long double value = parm[0]->data()->getLongDoubleVal(row, isNull);
+
+            if (isNull || (value < -1.0 || value > 1.0))
```

Comment:
I believe this conditional should match line 540. Domain of cosine can be less than 1 and greater than 1

Reply:
Correct.